core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -628,12 +565,12 @@ public void onConnectionClosedByServer(Status status) {
 
                 @Override
                 public void onInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, true);
+                  isIdle = false;
                 }
 
                 @Override
                 public void onNotInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, false);
+                  isIdle = true;
```

Comment:
This looks fishy. There are multiple transports, and even though one is no longer in use means none are in use.